### PR TITLE
Add coverage annotation and new tests for addTransceiver 

### PR DESCRIPTION
**Pull request by soareschen**
I updated the spec reference to the tip of the tree draft, as new PR for handling the direction argument was just merged recently.

The existing test code have not been modified. Coverage comment is added. I also added few basic tests for `MediaStreamTrack` argument and direction argument.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245224983)

# Firefox (nightly) #
Testing web-platform-tests at revision 0e7b84d888ddd8dfc0f56ddb07729df9b10aeafb
Using browser at version BuildID 20170612100241; SourceStamp 27cad9749cddf68e11fdd4e5d73dad84a8f8cf23
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6298/webrtc/RTCPeerConnection-addTransceiver.html">/webrtc/RTCPeerConnection-addTransceiver.html</a></summary>


|                                             Subtest                                             | Results |                                                   Messages                                                  |
|-------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------------------|
|                                                                                                 | OK      |                                                                                                             |
| `addTransceiver() with string argument as invalid kind should throw TypeError`                  | FAIL    | `assert_own_property: expected property "addTransceiver" missing`                                           |
| `addTransceiver('audio') should return an audio transceiver`                                    | FAIL    | `assert_own_property: expected property "addTransceiver" missing`                                           |
| `addTransceiver('video') should return a video transceiver`                                     | FAIL    | `assert_own_property: expected property "addTransceiver" missing`                                           |
| `addTransceiver() with direction sendonly should have result transceiver.direction be the same` | FAIL    | `pc.addTransceiver is not a function`                                                                       |
| `addTransceiver() with direction inactive should have result transceiver.direction be the same` | FAIL    | `pc.addTransceiver is not a function`                                                                       |
| `addTransceiver() with invalid direction should throw TypeError`                                | FAIL    | `assert_own_property: expected property "addTransceiver" missing`                                           |
| `addTransceiver(track) should return transceiver with sender.track be track`                    | FAIL    | `assert_own_property: Expect pc to have addTransceiver() method expected property "addTransceiver" missing` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245224984)

# Sauce (safari) #
Testing web-platform-tests at revision 0e7b84d888ddd8dfc0f56ddb07729df9b10aeafb
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6298/webrtc/RTCPeerConnection-addTransceiver.html">/webrtc/RTCPeerConnection-addTransceiver.html</a></summary>


|                                             Subtest                                             | Results |                 Messages                 |
|-------------------------------------------------------------------------------------------------|---------|------------------------------------------|
|                                                                                                 | OK      |                                          |
| `addTransceiver() with string argument as invalid kind should throw TypeError`                  | FAIL    | `Can't find variable: RTCPeerConnection` |
| `addTransceiver('audio') should return an audio transceiver`                                    | FAIL    | `Can't find variable: RTCPeerConnection` |
| `addTransceiver('video') should return a video transceiver`                                     | FAIL    | `Can't find variable: RTCPeerConnection` |
| `addTransceiver() with direction sendonly should have result transceiver.direction be the same` | FAIL    | `Can't find variable: RTCPeerConnection` |
| `addTransceiver() with direction inactive should have result transceiver.direction be the same` | FAIL    | `Can't find variable: RTCPeerConnection` |
| `addTransceiver() with invalid direction should throw TypeError`                                | FAIL    | `Can't find variable: RTCPeerConnection` |
| `addTransceiver(track) should return transceiver with sender.track be track`                    | FAIL    | `Can't find variable: RTCPeerConnection` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245224990)

# Chrome (unstable) #
Testing web-platform-tests at revision 0e7b84d888ddd8dfc0f56ddb07729df9b10aeafb

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/245224991)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 0e7b84d888ddd8dfc0f56ddb07729df9b10aeafb

File: webrtc/RTCPeerConnection-addTransceiver.html
Comment (rwaldron) [inline webrtc/RTCPeerConnection-addTransceiver.html]: Should this PR be blocked for merging until the changes are published in a new draft?

Comment (foolip) [inline webrtc/RTCPeerConnection-addTransceiver.html]: I think that as soon as a change is merged into the Git repo, tests can be merged. It's a repeated source of confusion and frustration that the published version doesn't match, see https://github.com/w3c/webrtc-pc/issues/942

Using the rawgit URL here seems fine since even after being merged, there's no way to refer to a specific revision.

**Comment by rwaldron** [comment]
@soareschen this PR needs to be rebased

**Comment by soareschen** [comment]
Hmm apparently I did some double work and forgot about #6249. Have rebased now.

**Comment by rwaldron** [comment]
@soareschen for some reason I'm not listed as a reviewer in the list above and therefore cannot merge this :| (otherwise I would)

**Review by rwaldron — APPROVED** [review]
I will keep track of the PRs that include coverage for spec prose that is not yet published.

**Pull request merged**
